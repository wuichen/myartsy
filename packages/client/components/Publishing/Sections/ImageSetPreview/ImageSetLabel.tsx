import { Flex, Sans, space } from "@artsy/palette"
import React from "react"
import styled from "styled-components"

import { IconImageSet } from "Components/Publishing/Icon/IconImageSet"
import { Media } from "Utils/Responsive"
import { ImageSetPreviewProps } from "./ImageSetPreview"

export const ImageSetLabel = (props: ImageSetPreviewProps) => {
  const {
    color,
    section: { images, title },
  } = props
  const label = images.length === 1 ? "Image" : "Images"
  const imageCount = `${images.length} ${label}`
  const primaryTitle = title ? title : imageCount
  const textColor = color !== "white" ? color : undefined

  return (
    <LabelWrapper alignItems="center" justifyContent="space-between">
      <Flex flexDirection="column" justifyContent="space-between">
        <SlideshowTitle
          size={["4", "5"]}
          weight="medium"
          pb={2}
          color={textColor}
        >
          {primaryTitle}
        </SlideshowTitle>

        <SlideshowCta>
          <Sans size={["2", "3"]} weight="medium" color={textColor}>
            View Slideshow
          </Sans>
          {title && (
            <Sans size={["2", "3"]} pl={20} color={textColor}>
              {imageCount}
            </Sans>
          )}
        </SlideshowCta>
      </Flex>

      <Media greaterThanOrEqual="sm">
        <IconContainer>
          <IconImageSet color={textColor} />
        </IconContainer>
      </Media>
    </LabelWrapper>
  )
}

export const IconContainer = styled.div`
  height: 45px;
  position: relative;
  margin-left: ${space(4)}px;
  text-align: right;

  > svg {
    height: 98%;
  }
`

export const LabelWrapper = styled(Flex)`
  padding: ${space(2)}px;
  width: 100%;
`

// exported for targeting from outside components
export const SlideshowTitle = styled(Sans)``
export const SlideshowCta = styled(Flex)``
