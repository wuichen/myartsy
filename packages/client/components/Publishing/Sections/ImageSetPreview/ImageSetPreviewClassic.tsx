import { avantgarde } from "Assets/Fonts"
import { ImagesData } from "Components/Publishing/Typings"
import React, { Component } from "react"
import styled, { StyledFunction } from "styled-components"
import { resize } from "../../../../Utils/resizer"
import { IconImageSet } from "../../Icon/IconImageSet"

interface LengthProps extends React.HTMLProps<HTMLDivElement> {
  imgLength: number
}

const Wrapper = styled.div`
  max-width: 580px;
  width: 100%;
  height: 150px;
  display: flex;
`
const Text = styled.div`
  ${avantgarde("s11")};
  line-height: 1.35em;
  margin: 0;
`
const Remaining = styled.div`
  min-width: 50px;
  padding: 10px;
  flex: 1;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border: 1px solid #e5e5e5;
  text-align: center;
`
const IconContainer = styled.div`
  width: 32px;
  margin-bottom: 10px;
  position: relative;
`
const container = 560
const containerHeight = 150
const iconContainer = 50
const margin = 10

export interface ImageSetPreviewClassicProps {
  images: ImagesData
}

interface ImageSetPreviewClassicState {
  visibleImages: number
}

export class ImageSetPreviewClassic extends Component<
  ImageSetPreviewClassicProps,
  ImageSetPreviewClassicState
> {
  constructor(props) {
    super(props)
    this.state = {
      visibleImages: this.getVisibleImages(this.props.images),
    }
  }

  getVisibleImages(images) {
    const widths = []
    let hidden = 0
    images.map((item, i) => {
      const adjustedWidth = (containerHeight * item.width) / item.height
      widths.push(adjustedWidth)
      const total = widths.reduce((a, b) => a + b, 0)
      const margins = widths.length * margin
      if (total + margins + iconContainer > container) {
        hidden = hidden + 1
      }
    })
    return widths.length - hidden
  }

  renderImages(images) {
    const items = images.slice(0, 4).map((item, i) => {
      const src = resize(item.image || item.url || "", { width: 500 })
      const alt = item.caption
        ? item.caption.replace(/<[^>]*>/g, "")
        : item.title || ""
      if (i < this.state.visibleImages) {
        return (
          <img
            key={"imageset-" + i}
            src={src}
            height={containerHeight}
            className="imageset-preview__image"
            style={{ marginRight: margin }}
            alt={alt}
          />
        )
      }
    }, this)
    return items
  }

  render() {
    const { images } = this.props

    return (
      <Wrapper className="imageset-preview">
        <div
          className="imageset-preview__container"
          style={{ display: "flex" }}
        >
          {this.renderImages(images)}
        </div>
        <Remaining className="imageset-preview__remaining">
          <IconContainer className="imageset-preview__icon-container">
            <IconImageSet />
            <Length
              className="imageset-preview__length"
              imgLength={images.length}
            >
              <Text>{images.length}</Text>
            </Length>
          </IconContainer>
          <Text className="imageset-preview__text">Enter Slideshow</Text>
        </Remaining>
      </Wrapper>
    )
  }
}

const Div: StyledFunction<LengthProps> = styled.div
const Length = Div`
  position: absolute;
  left: ${props => (props.imgLength > 9 ? "4px" : "8px")};
  top: 14px;
`
