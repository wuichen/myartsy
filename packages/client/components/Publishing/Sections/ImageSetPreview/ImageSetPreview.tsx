import { color as Color, Flex } from "@artsy/palette"
import { withFullScreen } from "Components/Publishing/Sections/FullscreenViewer/withFullScreen"
import { ImagesData } from "Components/Publishing/Typings"
import React, { ReactNode } from "react"
import styled from "styled-components"
import { resize } from "Utils/resizer"
import { ImageSetLabel } from "./ImageSetLabel"

type Layout = "mini" | "full"

export interface ImageSetSection {
  type: string
  images: ImagesData
  layout?: Layout
  title?: string
}

export interface ImageSetPreviewProps {
  color?: string
  section: ImageSetSection
  children?: ReactNode
  onViewFullscreen?: (index: number) => void
  fullscreenImages?: ImagesData
}

@withFullScreen
export class ImageSetPreview extends React.PureComponent<ImageSetPreviewProps> {
  getImageUrl() {
    const { images, layout } = this.props.section
    const image = images[0]
    const src = image.url ? image.url : image.image
    const width = layout === "full" ? 800 : 200

    return resize(src, { width })
  }

  onClick = () => {
    const {
      fullscreenImages,
      section: { images },
    } = this.props
    const slideshowIndex =
      fullscreenImages &&
      fullscreenImages.findIndex(img => {
        return img.type === "artwork"
          ? img.image === images[0].image
          : img.url === images[0].url
      })

    this.props.onViewFullscreen(slideshowIndex)
  }

  render() {
    const {
      children,
      color,
      section: { layout, title },
    } = this.props
    const alt = title || "Open Slideshow"
    const src = this.getImageUrl()

    if (layout === "full") {
      return (
        <ImageSetContainer>
          <FullLabel onClick={this.onClick}>
            <ImageSetLabel {...this.props} />
            {children}
          </FullLabel>
          <ImgContainer>
            <Img src={src} alt={alt} />
          </ImgContainer>
        </ImageSetContainer>
      )
    } else {
      return (
        <ImageSetContainer>
          <MiniWrapper alignItems="center" onClick={this.onClick} color={color}>
            <Img src={src} alt={alt} />
            <ImageSetLabel {...this.props} />
          </MiniWrapper>
        </ImageSetContainer>
      )
    }
  }
}

export const ImageSetContainer = styled.div`
  position: relative;
  width: 100%;
`

export const FullLabel = styled.div`
  position: absolute;
  bottom: 20px;
  left: 20px;
  min-height: 50px;
  width: auto;
  max-width: calc(100% - 80px);
  border-radius: 2px;
  background: rgba(255, 255, 255, 0.8);
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.3);
  cursor: pointer;

  &:hover {
    background: rgba(0, 0, 0, 0.6);
    color: white;

    path,
    polygon {
      fill: white;
    }
  }
`

export const Img = styled.img`
  height: auto;
  width: 100%;
`

export const ImgContainer = styled.div``

const MiniWrapper = styled(Flex)<{ color?: string }>`
  height: 100px;
  padding: 10px 0 10px 10px;
  border: 1px solid ${props => (props.color ? props.color : Color("black10"))};
  cursor: pointer;

  ${Img} {
    height: 100%;
    width: auto;
  }
`
