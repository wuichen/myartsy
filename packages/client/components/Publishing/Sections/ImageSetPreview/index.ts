import { ImageSetPreview } from "./ImageSetPreview"
export default ImageSetPreview
export * from "./ImageSetPreview"
