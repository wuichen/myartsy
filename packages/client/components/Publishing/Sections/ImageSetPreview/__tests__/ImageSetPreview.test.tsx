import {
  ImageSetFull,
  ImageSetMini,
} from "Components/Publishing/Fixtures/Components"
import { WrapperWithFullscreenContext } from "Components/Publishing/Fixtures/Helpers"
import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { ImageSetPreview } from "../ImageSetPreview"

const renderSnapshot = _props => {
  return renderer
    .create(WrapperWithFullscreenContext(<ImageSetPreview {..._props} />))
    .toJSON()
}
const props = { section: null }

it("renders a full image set properly", () => {
  props.section = ImageSetFull
  const imageset = renderSnapshot(props)
  expect(imageset).toMatchSnapshot()
})
it("renders a mini image set properly", () => {
  props.section = ImageSetMini
  const imageset = renderSnapshot(props)
  expect(imageset).toMatchSnapshot()
})
