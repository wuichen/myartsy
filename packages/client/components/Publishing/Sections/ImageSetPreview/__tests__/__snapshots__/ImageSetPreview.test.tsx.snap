// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a full image set properly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 16px;
  line-height: 26px;
  padding-bottom: 8px;
}

.c6 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
}

.c7 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 16px;
  padding-left: 20px;
}

.c8 {
  height: 45px;
  position: relative;
  margin-left: 40px;
  text-align: right;
}

.c8 > svg {
  height: 98%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  position: relative;
  width: 100%;
}

.c1 {
  position: absolute;
  bottom: 20px;
  left: 20px;
  min-height: 50px;
  width: auto;
  max-width: calc(100% - 80px);
  border-radius: 2px;
  background: rgba(255,255,255,0.8);
  box-shadow: 0 0 10px 0 rgba(0,0,0,0.3);
  cursor: pointer;
}

.c1:hover {
  background: rgba(0,0,0,0.6);
  color: white;
}

.c1:hover path,
.c1:hover polygon {
  fill: white;
}

.c9 {
  height: auto;
  width: 100%;
}

@media screen and (min-width:40em) {
  .c4 {
    font-size: 18px;
  }
}

@media screen and (min-width:40em) {
  .c4 {
    line-height: 30px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    font-size: 14px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    line-height: 24px;
  }
}

@media screen and (min-width:40em) {
  .c7 {
    font-size: 14px;
  }
}

@media screen and (min-width:40em) {
  .c7 {
    line-height: 24px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize={
            Array [
              "16px",
              "18px",
            ]
          }
        >
          The Work of Bruce M. Sherman
        </div>
        <div
          className="c5"
        >
          <div
            className="c6"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize={
              Array [
                "12px",
                "14px",
              ]
            }
          >
            View Slideshow
          </div>
          <div
            className="c7"
            fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
            fontSize={
              Array [
                "12px",
                "14px",
              ]
            }
          >
            3 Images
          </div>
        </div>
      </div>
      <div
        className="rrm-container rrm-greaterThanOrEqual-sm"
      >
        <div
          className="c8"
        >
          <svg
            className="image-set"
            viewBox="0 0 44 44"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
              fill="black"
              transform="translate(-6728.67 -507.72)"
            />
            <polygon
              fill="black"
              points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
            />
            <polygon
              fill="black"
              points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className=""
  >
    <img
      alt="The Work of Bruce M. Sherman"
      className="c9"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=800&quality=80"
    />
  </div>
</div>
`;

exports[`renders a mini image set properly 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 16px;
  line-height: 26px;
  padding-bottom: 8px;
}

.c8 {
  font-family: Unica77LLWebMedium,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
}

.c9 {
  font-family: Unica77LLWebRegular,'Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 16px;
  padding-left: 20px;
}

.c10 {
  height: 45px;
  position: relative;
  margin-left: 40px;
  text-align: right;
}

.c10 > svg {
  height: 98%;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 20px;
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  position: relative;
  width: 100%;
}

.c3 {
  height: auto;
  width: 100%;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100px;
  padding: 10px 0 10px 10px;
  border: 1px solid #E5E5E5;
  cursor: pointer;
}

.c1 .c2 {
  height: 100%;
  width: auto;
}

@media screen and (min-width:40em) {
  .c6 {
    font-size: 18px;
  }
}

@media screen and (min-width:40em) {
  .c6 {
    line-height: 30px;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    font-size: 14px;
  }
}

@media screen and (min-width:40em) {
  .c8 {
    line-height: 24px;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    font-size: 14px;
  }
}

@media screen and (min-width:40em) {
  .c9 {
    line-height: 24px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <img
      alt="The Work of Bruce M. Sherman"
      className="c2 c3"
      src="https://d7hftxdivxxvm.cloudfront.net?resize_to=width&src=https%3A%2F%2Fd32dm0rphc51dk.cloudfront.net%2F0aRUvnVgQKbQk5dj8xcCAg%2Flarger.jpg&width=200&quality=80"
    />
    <div
      className="c4"
    >
      <div
        className="c5"
      >
        <div
          className="c6"
          fontFamily={
            Object {
              "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
              "fontWeight": 500,
            }
          }
          fontSize={
            Array [
              "16px",
              "18px",
            ]
          }
        >
          The Work of Bruce M. Sherman
        </div>
        <div
          className="c7"
        >
          <div
            className="c8"
            fontFamily={
              Object {
                "fontFamily": "Unica77LLWebMedium, 'Helvetica Neue', Helvetica, Arial, sans-serif",
                "fontWeight": 500,
              }
            }
            fontSize={
              Array [
                "12px",
                "14px",
              ]
            }
          >
            View Slideshow
          </div>
          <div
            className="c9"
            fontFamily="Unica77LLWebRegular, 'Helvetica Neue', Helvetica, Arial, sans-serif"
            fontSize={
              Array [
                "12px",
                "14px",
              ]
            }
          >
            3 Images
          </div>
        </div>
      </div>
      <div
        className="rrm-container rrm-greaterThanOrEqual-sm"
      >
        <div
          className="c10"
        >
          <svg
            className="image-set"
            viewBox="0 0 44 44"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6760.67,551.72h-32v-32h32v32Zm-30-2h28v-28h-28v28Z"
              fill="black"
              transform="translate(-6728.67 -507.72)"
            />
            <polygon
              fill="black"
              points="38 37 36 37 36 8 7 8 7 6 38 6 38 37"
            />
            <polygon
              fill="black"
              points="44 31 42 31 42 2 13 2 13 0 44 0 44 31"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
